lib/Target/Mips/MipsSEISelLowering.cpp

```diff
@@ -2379,6 +2387,16 @@ SDValue MipsSETargetLowering::lowerINTRINSIC_WO_CHAIN(SDValue Op,
     EVT PtrVT = getPointerTy(DAG.getDataLayout());
     return DAG.getNode(MipsISD::ThreadPointer, DL, PtrVT);
   }
+  case Intrinsic::cheri_cap_address_get: {
```

Comment:
I wonder if we should expand this earlier in the IR and then re-run the pass that removes redundant capability operations.  We're probably missing some potential optimisations otherwise.